From d5112ddc9691c9ee593484135ef80a73f52560fd Mon Sep 17 00:00:00 2001
From: Vince Chang <vince_chang@aspeedtech.com>
Date: Fri, 18 Oct 2024 16:03:53 +0800
Subject: [PATCH] fix no postcodeIface

If the PFR mailbox is not on I2C4 or the address is not 0x38 on the I2C bus,
it cannot retrieve the provision state and create the postcode interface.
postcodeIface = "xyz.openbmc_project.State.Boot.Platform";

Signed-off-by: Vince Chang <vince_chang@aspeedtech.com>
---
 service/src/mainapp.cpp | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/service/src/mainapp.cpp b/service/src/mainapp.cpp
index 53a0147..f4357bf 100644
--- a/service/src/mainapp.cpp
+++ b/service/src/mainapp.cpp
@@ -710,12 +710,9 @@ int main()
 
     if (pfr::pfrConfigObject)
     {
-        pfr::pfrConfigObject->updateProvisioningStatus();
-        if (pfr::pfrConfigObject->getPfrProvisioned())
-        {
-            pfr::pfrPostcodeObject =
-                std::make_unique<pfr::PfrPostcode>(server, conn);
-        }
+        // Create PFR post code interface
+        pfr::pfrPostcodeObject =
+            std::make_unique<pfr::PfrPostcode>(server, conn);
     }
 
     conn->request_name("xyz.openbmc_project.PFR.Manager");
